src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -302,17 +313,32 @@ where
                         .observe(builder_output.epoch_count as _);
                 }
             }
-            self.sst_outputs.push(SplitTableOutput {
-                upload_join_handle: builder_output.writer_output,
-                sst_info: builder_output.sst_info,
-            });
+
+            self.concurrent_upload_join_handle
+                .push_back(builder_output.writer_output);
+
+            self.sst_outputs.push(builder_output.sst_info);
+
+            if let Some(concurrent_uploading_sst_count) = self.concurrent_uploading_sst_count
+                && self.concurrent_upload_join_handle.len() >= concurrent_uploading_sst_count
+            {
+                let join_handle = self.concurrent_upload_join_handle.pop_front().unwrap();
+                join_handle
+                    .verbose_instrument_await("upload")
+                    .await
+                    .map_err(HummockError::sstable_upload_error)??;
+            }
```

Comment:
The logic here polls the join handles in FIFO order. Will it be better if we use something like `FutureUnordered` instead?

Reply:
Maybe `select_all` is better but I think in most case, the first sst enter queue shall finish uploading in order